### ISSUE TITLE
bug: Updating docker broke lambdas in older versions of localstack

### ISSUE THREAD

**Issue by adamdavis40208** [localstack/localstack#6650]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

So this is a weird one. We recently updated our machine image in circleCI from an older tag (Ubuntu 20.05, Docker v20.10.7, Docker-compose v1.29.2) to a new tag (#Ubuntu 22.04, Docker v20.10.14, Docker Compose v2.4.1).

Pretty suddenly we noticed failures around repos using localstack 0.11 and 0.12 and lambdas. Bumping them to a newer version of localstack 0.14 fixes it. 

Updating docker-desktop locally let me reproduce this issue. 

I'm just very simply spinning up localstack, and then using aws cli to create a small python lambda and execute it. 

```localstack-11_1  | Exception: Lambda process returned error status code: 1. Result: {"errorType":"Runtime.HandlerNotFound","errorMessage":"Handler 'handler' missing on module 'code'"}.```

vs localstack 14:

```localstack-14_1  | 2022-08-11T05:54:26.076:DEBUG:localstack.utils.container_utils.docker_sdk_client: Copying file /tmp/localstack/zipfile.20620f18/. into ba13dd72a267c0dd67f94ba13447f4023e9b91c1c311b0719880caec77428b92:/var/task
localstack-14_1  | 2022-08-11T05:54:26.243:DEBUG:localstack.utils.container_utils.docker_sdk_client: Starting container ba13dd72a267c0dd67f94ba13447f4023e9b91c1c311b0719880caec77428b92
lambda-tests_1   | "I worked!"
lambda-tests_1   | {
lambda-tests_1   |     "StatusCode": 200,
lambda-tests_1   |     "LogResult": "",
lambda-tests_1   |     "ExecutedVersion": "$LATEST"
lambda-tests_1   | }
janus-localstack_lambda-tests_1 exited with code 0
```

Does anyone have any idea why an older localstack would fail to create lambdas on a new version of docker or docker-compose?

I can throw up a sample repo if people want to see it in action.

### Expected Behavior

Spinning up a very simple lambda in a zip file should just work. 

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

docker-compose

```version: '2.2'
networks:
    docker-compose-network:
        name: docker-compose-network
        driver: bridge
services:
  lambda-tests:
    stdin_open: true
    build:
      context: .
      dockerfile: Dockerfile.callslambda
    networks:
      - docker-compose-network
    depends_on:
        - localstack-11
        - localstack-14

  localstack-11:
      image: localstack/localstack:0.11.4
      networks:
        - docker-compose-network
      environment:
          - SERVICES=lambda:4574,sns:4575,sqs:4576,s3:4572,ssm:4583,dynamodb:4569
          - LAMBDA_DOCKER_NETWORK=docker-compose-network
          - DEBUG=true
          - LAMBDA_EXECUTOR=docker
          - DOCKER_HOST=unix:///var/run/docker.sock
          - DEFAULT_REGION=us-west-2
          - HOSTNAME_EXTERNAL=localstack
      volumes:
          - "/var/run/docker.sock:/var/run/docker.sock"
      ports:
          - "4566" #edge-port
  localstack-14:
      image: localstack/localstack:0.14.5
      networks:
        - docker-compose-network
      environment:
          - SERVICES=lambda:4574,sns:4575,sqs:4576,s3:4572,ssm:4583,dynamodb:4569
          - LAMBDA_DOCKER_NETWORK=docker-compose-network
          - DEBUG=true
          - LAMBDA_EXECUTOR=docker
          - DOCKER_HOST=unix:///var/run/docker.sock
          - DEFAULT_REGION=us-west-2
          - AWS_DEFAULT_REGION=us-west-2
          - HOSTNAME_EXTERNAL=localstack
      volumes:
          # for lambda to use
          - "/var/run/docker.sock:/var/run/docker.sock"
      ports:
          - "4566" #edge-port
```

Dockerfile.callslambda:

```FROM amazon/aws-cli

COPY calllambda.sh /calllambda.sh

COPY code.zip /tmp/code.zip

CMD /calllambda.sh

ENTRYPOINT /calllambda.sh
```

Calllambda.sh:

```
#!/bin/bash
sleep 30
aws configure set default.region us-west-2
aws configure set aws_access_key_id 'YOUR_ACCESS_KEY'
aws configure set aws_secret_access_key 'YOUR_SECRET_KEY'
aws lambda list-functions --region us-west-2 --endpoint-url http://localstack-11:4566
aws lambda create-function --region us-west-2 --function-name func1 --runtime python3.8 --role r1 --handler code.handler --timeout 30 --zip-file fileb:///tmp/code.zip --endpoint-url http://localstack-11:4566
aws lambda list-functions --region us-west-2 --endpoint-url http://localstack-11:4566
aws lambda invoke --region us-west-2 --function-name func1 /dev/stdout --endpoint http://localstack-11:4566

aws lambda list-functions --region us-west-2 --endpoint-url http://localstack-14:4566
aws lambda create-function --region us-west-2 --function-name func1 --runtime python3.8 --role r1 --handler code.handler --timeout 30 --zip-file fileb:///tmp/code.zip --endpoint-url http://localstack-14:4566
aws lambda list-functions --region us-west-2 --endpoint-url http://localstack-14:4566
aws lambda invoke --region us-west-2 --function-name func1 /dev/stdout --endpoint http://localstack-14:4566
```

#### Lambda handler 

```Code.py

def handler(event, context):
    """Lambda handler that will get invoked by the LocalStack runtime"""
    print(event)
    return "I worked!"

if __name__ == "__main__":
    handler({}, {})
```

### Environment

```markdown
- OS: mac
- Docker: Engine: 20.10.17 Compose: 1.29.2
- LocalStack: 0.11.4 & 0.14.5
```

**Comment by adamdavis40208** [comment]
Just added the docker-compose stuff, forgot to add the full file. Thanks!

**Comment by ewascent** [comment]
@chezsmithy  if you are bored...

**Comment by chezsmithy** [comment]
I'm wondering if it has something to do with this issue. https://github.com/docker-library/ruby/issues/351

The older versions of localstack have an older version of the docker client. The lambda executor when using docker mode uses the docker client to fire up the image into the same context the local machine is running in. It's pulling latest lambda machine images which might just be later alpine causing an incompatibility.

**Comment by dfangl** [comment]
Lambda machine images are in general amazonlinux 1/2, so this should not affect this. It also seems the other way around, so a new docker is breaking an old image.

On my machine, with newest docker: `Docker version 20.10.21, build baeda1f82a` I actually cannot reproduce it, lambda invocation works as expected even with 0.11.

Overall, I am not sure we can do much here. In general, we do not support those old versions, especially 0.11 has some very significant security issues, so I would discourage using them in any case.

**Comment by dfangl** [comment]
I just tested the given code again, and lambda invocations worked for me on docker version `20.10.22, build 3a2c30b63a` with LocalStack 0.11.4.
I just had to add the mount `/tmp/localstack:/tmp/localstack` in the docker compose file, then it would find the handler.
In later versions, the default would use docker cp to get the archive into the container, which is why it will work with 0.14.

Nevertheless, I will close this, as a viable and recommended workaround (updating to 0.14 or higher, I would recommend our current latest 1.3) is available!
